js/client/modules/@arangodb/testutils/instance-manager.js

```diff
@@ -182,6 +183,40 @@ class instanceManager {
     }
     fs.removeDirectoryRecursive(this.rootDir, true);
   }
+
+  getMemLayout () {
+    if (this.options.memory !== undefined) {
+      if (this.options.cluster) {
+        // Distribute 10 % agency, 20% coordinator, 70% dbservers
+        this.memlayout[instanceRole.agent] = Math.round(this.options.memory * (10/100) / this.options.agencySize);
+        this.memlayout[instanceRole.dbServer] = Math.round(this.options.memory * (70/100) / this.options.dbServers);
+        this.memlayout[instanceRole.coordinator] = Math.round(this.options.memory * (20/100) / this.options.coordinators);
+      } else if (this.options.activefailover) {
+        // Distribute 20% agency, 80% singles
+        this.memlayout[instanceRole.agent] = Math.round(this.options.memory * (20/100) / this.options.agencySize);
+        this.memlayout[instanceRole.failover] = Math.round(this.options.memory * (80/100) / this.options.singles);
+      } else if (this.options.agency) {
+        this.memlayout[instanceRole.agent] = Math.round(this.options.memory / this.options.agencySize);
+      } else {
+        this.memlayout[instanceRole.single] = Math.round(this.options.memory / this.options.singles);
+      }
+    } else {
+      if (this.options.cluster) {
+        // Distribute 10 % agency, 20% coordinator, 70% dbservers
+        this.memlayout[instanceRole.agent] = 0;
+        this.memlayout[instanceRole.dbServer] = 0;
+        this.memlayout[instanceRole.coordinator] = 0;
+      } else if (this.options.activefailover) {
+        // Distribute 20% agency, 80% singles
+        this.memlayout[instanceRole.agent] = 0;
+        this.memlayout[instanceRole.failover] = 0;
+      } else if (this.options.agency) {
+        this.memlayout[instanceRole.agent] = 0;
+      } else {
+        this.memlayout[instanceRole.single] = 0;
+      }
+    }
```

Comment:
What is the expected behavior in this case? Is it an old behavior?

Also comments seems to be wrong here

Reply:
yes, old behaviour - no memory limit set. 
Yes, probably should remove the comments.